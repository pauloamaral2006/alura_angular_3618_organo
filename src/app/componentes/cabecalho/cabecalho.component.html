<header class="banner">
  <img [src]="src" [alt]="alt" />
</header>
