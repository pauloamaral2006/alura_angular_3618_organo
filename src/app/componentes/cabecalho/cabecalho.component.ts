import { Component, Input } from '@angular/core';

@Component({
  selector: 'app-cabecalho',
  standalone: true,
  imports: [],
  templateUrl: './cabecalho.component.html',
  styleUrl: './cabecalho.component.css'
})
export class CabecalhoComponent {
  @Input() src = '';
  @Input() alt = '';
}
