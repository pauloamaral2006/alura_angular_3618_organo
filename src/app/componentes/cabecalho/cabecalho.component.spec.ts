import { ComponentFixture, TestBed } from '@angular/core/testing';
import { CabecalhoComponent } from './cabecalho.component';

describe('CabecalhoComponent', () => {
  let component: CabecalhoComponent;
  let fixture: ComponentFixture<CabecalhoComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [CabecalhoComponent], // standalone
    }).compileComponents();

    fixture = TestBed.createComponent(CabecalhoComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('deveria ser criado', () => {
    expect(component).toBeTruthy();
  });

  it('deveria definir as propriedades alt e src', () => {
    expect(component.alt).toBeDefined();
    expect(component.src).toBeDefined();
  });

  it('deveria renderizar o conteÃºdo baseado nas propriedades src e alt', () => {
    component.src = 'https://example.com/test-image.jpg';
    component.alt = 'Imagem teste';
    expect(component).toMatchSnapshot();
  });
});
