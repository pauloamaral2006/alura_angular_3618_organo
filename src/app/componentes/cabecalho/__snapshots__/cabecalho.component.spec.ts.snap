// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CabecalhoComponent deveria renderizar o conte√∫do baseado nas propriedades src e alt 1`] = `
CabecalhoComponent {
  "__ngContext__": 4,
  "alt": "Imagem teste",
  "src": "https://example.com/test-image.jpg",
}
`;
